### PR TITLE
Configure Travis tests for Python versions

### PR DESCRIPTION
**Pull request by naoyak**
Description of changeset: 
Travis scripts were not actually using the Python versions specified in the build matrix (`conda` was installing 3.7 for the Python 3 build).

Test Plan: 
Run CI

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

### CODE REVIEW

File: tests/config_server.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # This file is only needed for setting up a dedicated server configuration
 # with support for extracting username information and emailing users.
 
-SERVER_NAME = 'localhost'
+SERVER_NAME = 'localhost.localdomain'
```
Comment (matthewwardrop) [inline tests/config_server.py]: Why this change? I think this will lead to platform specific behaviour, no?

File: appveyor.yml [hunk 2]
```diff
@@ -7,9 +7,9 @@ environment:
       PYTHON_ARCH: "32"
       MINICONDA: C:\Miniconda
 
-    - PYTHON: "C:\\Python34"
-      PYTHON_VERSION: "3.4.1"
-      PYTHON_ARCH: "32"
+    - PYTHON: "C:\\Python37-x64"
+      PYTHON_VERSION: "3.7.0"
```
Comment (matthewwardrop) [inline appveyor.yml]: We only explicit claim support for Python 3.6, which may change soon; but in the mean time, let's target 3.6.6 here.

File: tests/config_server.py (hunk 1 above)
Comment (naoyak) [reply tests/config_server.py]: @matthewwardrop so I noticed that the test runs have a bunch of [Flask warnings](https://travis-ci.org/airbnb/knowledge-repo/jobs/433841809#L1065-L1109), and this makes them go away. This config is only for running the test suite, right?

Comment (matthewwardrop) [reply tests/config_server.py]: @naoyak Yep... I was worried it might break Windows... but it didn't! Merging!
